### PR TITLE
feat: implement topic manager

### PR DESCRIPTION
**Pull request by fabian-romero-jalafund**
## Related Ticket
- [LU 10](https://trello.com/c/F5YbfxqF)

## Description

This PR contains the implementation class for topic management.

## Type of Change

Put an `x` in all the boxes that apply:

- [X] ✨ New Feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug Fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking Change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code Refactor
- [ ] ✅ Build Configuration Change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

## Pull Request Checklist

Put an `x` in all the boxes that apply:

- [x] My commit messages are detailed.
- [x] My code follows the code style of this project.
- [x] No existing features have been broken without good reason.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests pass.

### CODE REVIEW

File: app/src/main/java/salesian/university/broker/TopicManager.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package salesian.university.broker;
+
+import java.util.*;
```
Comment (teran-joseluis) [inline app/src/main/java/salesian/university/broker/TopicManager.java]: It'd be good if you just import those objects you use into the class and not import all.

File: app/src/main/java/salesian/university/broker/TopicManager.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    public TopicManager() {
+        this.subscriptions = new HashMap<>();
+    }
+
+    public void createTopic(String topic) {
+        subscriptions.putIfAbsent(topic.toLowerCase(), new ArrayList<>());
+    }
+
+    public void addSubscriber(String topic, String consumerUrl) {
+        String topicLowerCase = topic.toLowerCase();
```
Comment (teran-joseluis) [inline app/src/main/java/salesian/university/broker/TopicManager.java]: Dont' you think we can have a utils class for converting topic into lower case? I see you've implemented in different lines.

File: app/src/main/java/salesian/university/broker/TopicManager.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        if (!subscriptions.containsKey(topicLowerCase)) {
+            throw new IllegalArgumentException("Topic does not exist: " + topic);
+        }
+        subscriptions.get(topicLowerCase).add(consumerUrl);
+    }
+
+    public List<String> getSubscribers(String topic) {
+        return subscriptions.getOrDefault(topic.toLowerCase(), Collections.emptyList());
```
Comment (teran-joseluis) [inline app/src/main/java/salesian/university/broker/TopicManager.java]: Why are you using `Collections.emptyList()`? Won't it return always an empty list?

**Review by teran-joseluis — CHANGES_REQUESTED** [review]
Don't forget to make update of your branch from develop.

**Comment by teran-joseluis** [comment]
Don't forget to update also your PR title using our convention, otherwise pipelines won't let you pass.

File: app/src/main/java/salesian/university/broker/TopicManager.java (hunk 3 above)
Comment (fabian-romero-jalafund) [reply app/src/main/java/salesian/university/broker/TopicManager.java]: It jus returns empty list when no topic key is found

File: app/src/main/java/salesian/university/broker/TopicManager.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
+package salesian.university.broker;
+
```
Comment (teran-joseluis) [inline app/src/main/java/salesian/university/broker/TopicManager.java:2]: Please remove this extra empty line.

File: app/src/test/java/salesian/university/broker/TopicManagerTest.java [hunk 5]
```diff
@@ -0,0 +1,91 @@
+package salesian.university.broker;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (teran-joseluis) [inline app/src/test/java/salesian/university/broker/TopicManagerTest.java]: Just import those functions you're using please.

File: app/src/test/java/salesian/university/helpers/StringManagerTest.java [hunk 6]
```diff
@@ -0,0 +1,59 @@
+package salesian.university.helpers;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (teran-joseluis) [inline app/src/test/java/salesian/university/helpers/StringManagerTest.java:5]: Same here please.

**Review by teran-joseluis — CHANGES_REQUESTED** [review]
Review theses final comments and make an update of your branch from develop as result I cannot see the pipelines status.

**Review by teran-joseluis — APPROVED** [review]
LGTM

**Pull request merged**